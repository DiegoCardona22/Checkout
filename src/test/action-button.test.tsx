// @packages
import { shallow, mount } from 'enzyme';

// @scripts
import ActionButton from 'components/atoms/action-button';

describe('Testing the component <ActionButton />', () => {
  const id = 'action-button';
  const label = 'Action';
  const startIcon = 'local_airport';
  const onClick = jest.fn();
  let wrapper = shallow(<ActionButton id={id} label={label} />);

  beforeEach(() => {
    wrapper = shallow(<ActionButton id={id} label={label} />);
  });

  it('Should render the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should show the label', () => {
    const localWrapper = mount(<ActionButton id={id} label={label} />);
    expect(localWrapper).toMatchSnapshot();
  });

  it('Should show the icon', () => {
    const localWrapper = mount(<ActionButton id={id} label={label} startIcon={startIcon} />);
    expect(localWrapper).toMatchSnapshot();
  });

  it('Should the click', () => {
    const localWrapper = mount(<ActionButton id={id} label={label} onClick={onClick} />);
    localWrapper.find('button').simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
});
