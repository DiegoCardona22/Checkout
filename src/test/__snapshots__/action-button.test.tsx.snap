// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the component <ActionButton /> Should render the component correctly 1`] = `ShallowWrapper {}`;

exports[`Testing the component <ActionButton /> Should show the icon 1`] = `ReactWrapper {}`;

exports[`Testing the component <ActionButton /> Should show the label 1`] = `ReactWrapper {}`;
